### DISCUSSION TITLE
Upload new metamodel

### DISCUSSION THREAD

**Discussion by marekstasik** [eclipse-sirius/sirius-components#2439] [General]
Hello,
Do you know how to upload my metamodel in Sirius Web? I create my metamodel and I do not know how upload it in Sirius Web. I followed the complete build instructions: https://github.com/eclipse-sirius/sirius-web 
What should I do next?

**Comment by AxelRICHARD** [comment]
Hello,

Assuming your metamodel is an ecore metamodel, then you should register it like the Flow ecore metamodel used in the `sirius-web-sample-application`.
The Flow ecore metamodel is registered in `SampleEMFConfiguration.java` (https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/backend/sirius-web-sample-application/src/main/java/org/eclipse/sirius/web/sample/configuration/SampleEMFConfiguration.java).

Regards,

**Reply by marekstasik** [reply]
Hello,
I have this metamodel https://github.com/marekstasik/metamodel/tree/my-new-branch/org.eclipse.sirius.sample.basicfamily/model
How can I change the SampleEMFConfiguration.java file that the metamodel working in Sirius Web?

**Reply by AxelRICHARD** [reply]
Hello,

As I said in my previous answer, you should do the same for your metamodel than for Flow:

```
@Configuration
public class SampleEMFConfiguration {
    @Bean
    public AdapterFactory familyAdapterFactory() {
        return new FamilyItemProviderAdapterFactory();
    }

    @Bean
    public EPackage familyEPackage() {
        return FamilyPackage.eINSTANCE;
    }
}
```

Furthermore, you should add your metamodel as a maven dependency of `sirius-web-sample-application` in https://github.com/eclipse-sirius/sirius-web/blob/master/packages/sirius-web/backend/sirius-web-sample-application/pom.xml
This means your metamodel project (and the .edit project associated) should be bundled as maven projects.
You can see what has been done for Flow metamodel (https://github.com/ObeoNetwork/Flow-Designer/tree/ocp) to help you to produce maven artifacts for your metamodel.

Regards,

**Reply by marekstasik** [reply]
Hello,
How can I bundle my metamodel project as maven projects? I convert my org.eclipse.sirius.sample.basicfamily folder to maven project 
![Screenshot from 2023-10-24 20-32-00](https://github.com/eclipse-sirius/sirius-web/assets/127620600/35f5bc2b-5688-4c0e-9558-6ddb1621fbca)
Then create the pom.xml file, but I can't execute  _mvn clean install -f pom.xml_ because show me the error
![image](https://github.com/eclipse-sirius/sirius-web/assets/127620600/43d6bc9c-e866-48c3-a9a6-b6ac6f92d73c)
What I should do it?
